weave/artifact_wandb.py

```diff
@@ -503,9 +516,16 @@ def add_artifact_reference(self, uri: str) -> None:
         # The URI for a get node might look like: wandb-artifact:///<entity>/<project>/test8:latest/obj
         # but the SDK add_reference call requires something like:
         # wandb-artifact://41727469666163743a353432353830303535/obj.object.json
-        assert uri.startswith("wandb-artifact:///")
-        uri_parts = WeaveWBArtifactURI.parse(uri)
-        uri_path = f"{uri_parts.entity_name}/{uri_parts.project_name}/{uri_parts.name}:{uri_parts.version}"
+        assert uri.startswith("wandb-artifact:///") or uri.startswith(
+            "wandb-artifact-by-id:///"
+        )
+        try:
+            uri_parts = WeaveWBArtifactURI.parse(uri)
+            uri_path = f"{uri_parts.entity_name}/{uri_parts.project_name}/{uri_parts.name}:{uri_parts.version}"
+        except errors.WeaveInternalError:
+            uri_by_id_parts = WeaveWBArtifactByIDURI.parse(uri)
+            uri_path = f"{uri_by_id_parts.path_root}/{uri_by_id_parts.artifact_id}/{uri_by_id_parts.name}:{uri_by_id_parts.version}"
+        # todo: how to handle this? this queries the public api
```

Comment:
is this todo resolved?

Reply:
no, thats actually something i was hoping to get some help on. :( thanks for pointing it out.

Reply:
Although i don't think we'll reach this error state since i dont see how a WeaveWBArtifactByIDURI instance could be initiated via add_reference() since the only place its happening is via artifact_version_file_ops(), but not sure how to handle this.

Reply:
in this case we probably want to throw if it is an ID-based URL, then?